src/components/connection_handler/src/connection.cc

```diff
@@ -159,6 +160,7 @@ bool Connection::AddNewService(uint8_t session_id,
   Service* service = session.FindService(service_type);
   // if service already exists
   if (service) {
+    *service_exists = true;
```

Comment:
@tkireva IMO here we can easily crash if the caller sets service_exists to NULL, please sanity check the parameter first.